scalableminds/webknossos
# Enhance tree search functionallity

This PR changes the search functionality in the trees tab view. It changes the search to a browser-like search where you can cycle through the found results via buttons or via pressing enter. The searched options are the name of the trees and the tree groups and the search should be case insensitive.

### URL of deployed dev instance (used for testing):
- [https://enhancetreesearchfunctionallity.webknossos.xyz](https://enhancetreesearchfunctionallity.webknossos.xyz)

### Steps to test:
- open a skeleton or hybrid tracing
- create some trees and groups
- use the search button or the shortcut shift + ctrl + f to open the search popover
- opening the popover should automatically focus the search input of the popover
- enter some keywords to search for
- if there is no result, hitting enter and using the buttons should not select a new tree or group as active and the text should be colored red
- when having multiple results, pressing enter or one of the buttons on the right side should cycle through the possible results without closing the search popover
- the popover can be closed with pressing escape, clicking somewhere else (not on the popover) or using the close button on the right side of the popover
- reopening the popover should keep the previously entered search sequence and index of "active search result" so that the user can continue to cycle through the results

### Issues:
- fixes #3826 #3708

------
